src/ObjectUtils/include/ObjectUtils/ReadMaps.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef OBJECT_UTILS_READ_MAPS_H_
+#define OBJECT_UTILS_READ_MAPS_H_
+
+#ifdef __linux
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include <filesystem>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include "GrpcProtos/module.pb.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_object_utils {
+
+// Represents an entry in /proc/[pid]/maps.
+class LinuxMemoryMapping {
+ public:
+  LinuxMemoryMapping(uint64_t start_address, uint64_t end_address, uint64_t perms, uint64_t offset,
+                     uint64_t inode, std::string pathname)
+      : start_address_{start_address},
+        end_address_{end_address},
+        perms_{perms},
+        offset_{offset},
+        inode_{inode},
+        pathname_{std::move(pathname)} {}
+
+  [[nodiscard]] uint64_t start_address() const { return start_address_; }
+  [[nodiscard]] uint64_t end_address() const { return end_address_; }
+  [[nodiscard]] uint64_t perms() const { return perms_; }
+  [[nodiscard]] uint64_t offset() const { return offset_; }
+  [[nodiscard]] uint64_t inode() const { return inode_; }
+  [[nodiscard]] const std::string& pathname() const { return pathname_; }
+
+ private:
+  uint64_t start_address_;
+  uint64_t end_address_;
+  uint64_t perms_;
+  uint64_t offset_;
+  uint64_t inode_;
+  std::string pathname_;
+};
+
+ErrorMessageOr<std::vector<LinuxMemoryMapping>> ReadMaps(pid_t pid);
+
+[[nodiscard]] std::vector<LinuxMemoryMapping> ReadMaps(std::string_view proc_pid_maps_content);
```

Comment:
So why is this method not returning an ErrorMessageOr? If I read the tests correctly, this is what it did before.

Reply:
Yeah, that was a bit confusing before. It could and can never happen. `ReadMaps(pid_t pid)` returns an error only if it can't read `/proc/[pid]/maps`, while this function is never encountering any error, it just skips malformed lines.